src/main/java/forestry/core/network/packets/PacketFXSignal.java

```diff
@@ -80,10 +77,9 @@ public void onPacketData(PacketBufferForestry data, EntityPlayer player) throws
 			BlockPos pos = data.readBlockPos();
 			VisualFXType visualFX = VisualFXType.values()[data.readByte()];
 			SoundFXType soundFX = SoundFXType.values()[data.readByte()];
-			Block block = Block.getBlockById(data.readVarInt());
-			IBlockState blockState = block.getStateFromMeta(data.readVarInt());
-
 			World world = player.world;
+			IBlockState blockState = world.getBlockState(pos);
```

Comment:
I'm not sure how bad this is for performance but I would say that looking for an ID in the block registry is probably better than searching the world for a block at a specific position.

Reply:
Why have you changed it if the other way was better ?

Reply:
The performance difference here is not a concern, but there is a possible race condition when we place/break a block and immediately fire some FX packets. The client may not have updated the world to match the server when it gets this packet, so it could get the wrong blockState here.